amazon-msk/cdktf/secure-public-access/zilla.yaml.mustache

```diff
@@ -0,0 +1,99 @@
+name: {{name}}
+vaults:
+  secure:
+    type: {{#useAcm}}aws-acm{{/useAcm}}{{^useAcm}}aws-secrets{{/useAcm}}
+telemetry:
+  metrics:
+    - stream.active.received
+    - stream.active.sent
+    - stream.opens.received
+    - stream.opens.sent
+    - stream.data.received
+    - stream.data.sent
+    - stream.errors.received
+    - stream.errors.sent
+    - stream.closes.received
+    - stream.closes.sent
+  exporters:
+    stdout_logs_exporter:
+      type: stdout
+{{#cloudwatch}}
+    aws0:
+      type: aws-cloudwatch
+      options:
+        logs:
+          group: {{cloudwatch.logs.group}}
+          stream: events
+        metrics:
+          namespace: {{cloudwatch.metrics.namespace}}
+{{/cloudwatch}}
+bindings:
+  tcp_server:
+    type: tcp
+    kind: server
+    options:
+      host: 0.0.0.0
+      port: {{publicPort}}
+    telemetry:
+      metrics:
+        - stream.*
+    exit: tls_server
+  tls_server:
+    type: tls
+    kind: server
+    vault: secure
+    options:
+      keys:
+        - {{publicTlsCertificateKey}}
+      {{#mTLS}}
+      trust:
+        - {{publicCertificateAuthority}}
+      {{/mTLS}}
+    telemetry:
+      metrics:
+        - stream.*
+    routes:
+      - exit: kafka_proxy
+        when:
+          - authority: '{{publicWildcardDNS}}'
+  kafka_proxy:
+    type: kafka-proxy
+    kind: proxy
+    options:
+      external:
+        host: '{{externalHost}}'
+        port: {{publicPort}}
+      internal:
+        host: '{{internalHost}}'
+        port: {{mskPort}}
+    telemetry:
+      metrics:
+        - stream.*
+    exit: tls_client
+  tls_client:
+    type: tls
+    kind: client
+    vault: secure
+    options:
+      {{#mTLS}}
+      signers:
+        - {{mskCertificateAuthority}}
+      {{/mTLS}}
+      trustcacerts: true
+    telemetry:
+      metrics:
+        - stream.*
+    exit: tcp_client
+  tcp_client:
+    type: tcp
+    kind: client
+    options:
+      host: '*'
+      port: {{mskPort}}
+    telemetry:
+      metrics:
+        - stream.*
+    routes:
+      - when:
+        - authority: '{{mskWildcardDNS}}'
+
```

Comment:
`msk` and `public` seem like they would work better as a structure instead too, similar to `cloudwatch` approach.